### ISSUE TITLE
Legal WAD filenames are incomplete

### ISSUE THREAD

**Issue by id-mikerubits** [MTrop/DoomStruct#6]
```
java.lang.IllegalArgumentException: The provided entry name, "^BRICKS", is invalid. It must be up to 8 characters long; all characters must be A-Z (uppercase only), 0-9, and [ ] - _ plus the backslash \.
```

This is when trying to load and merge a valid WAD file in the `wadmerge` utility.

**Comment by MTrop** [comment]
Thanks, @id-mikerubits. I'll need to figure out _exactly_ what an entry allows.

**Comment by MTrop** [comment]
Fixed in commit 4ddae4d1a102f88e5af1e3f7243de7750614ee1b
